polyfactory/factories/base.py

```diff
@@ -219,6 +220,28 @@ def __init_subclass__(cls, *args: Any, **kwargs: Any) -> None:  # noqa: C901
         if cls.__set_as_default_factory_for_type__:
             BaseFactory._factory_type_mapping[cls.__model__] = cls
 
+    @classmethod
+    def _infer_model_type(cls: type[F]) -> type[T] | None:
+        """Return model type inferred from class declaration.
+        class Foo(ModelFactory[MyModel]):  # <<< MyModel
+            ...
+
+        If more than one base class and/or generic arguments specified return None.
+
+        :returns: Inferred model type or None
+        """
+
+        factory_bases: Iterable[type[BaseFactory[T]]] = (
+            b for b in get_original_bases(cls) if get_origin(b) and issubclass(get_origin(b), BaseFactory)
+        )
+        generic_args: Sequence[type[T]] = [
+            arg for factory_base in factory_bases for arg in get_args(factory_base) if not isinstance(arg, TypeVar)
+        ]
+        if len(generic_args) != 1:
+            return None
+
+        return generic_args[0]
```

Comment:
```suggestion
        return None if len(generic_args) != 1 else generic_args[0]
```

Reply:
I don't like inline IF's much because IMHO it makes the code less readable. I can see the first if condition clearly and can understand it unambiguously. I can see the return statement then as well, and it also looks really clearly. 
And I should confess that the inline IF makes me think in this case -- I realise that I have to parse it in mind.   
But! It's not a problem for me to fix it and I'll do it if you insist.

Reply:
No big deal 😄

Reply:
Anyway, thank you for noticing it. I didn't know what sourcery is and didn't pay attention to refactoring. Now I realised how it works =)